system/jlib/jexcept.cpp

```diff
@@ -66,11 +69,141 @@ const char * querySeverityString(ErrorSeverity errorSeverity)
         return errorSeverityString[SeverityUnknown];
 }
 
+
+static std::unordered_map<std::string, ExceptionInterceptClass> eClassTypes = {
+    { "string", ExceptionInterceptClass::eString },
+    { "errno", ExceptionInterceptClass::eErrno},
+    { "os", ExceptionInterceptClass::eOs},
+    { "socket", ExceptionInterceptClass::eSocket}
+};
+class CExceptionInterceptor : implements IExceptionIntercept
+{
+    std::unordered_map<unsigned __int64, InterceptHandler> interceptHandlers;
+    StringBuffer debugDir;
+    void ensureDebugDir()
+    {
+        if (debugDir.length())
+            return;
+#ifdef _CONTAINERIZED
+        StringBuffer planeName;
+        if (!getDefaultPlane(planeName, "@debugPlane", "debug"))
+        {
+            WARNLOG("Exception handlers configured, but debug plane is missing");
+            return;
+        }
+        Owned<IPropertyTree> plane = getStoragePlane(planeName);
+        assertex(plane);
+        verifyex(plane->getProp("@prefix", debugDir));
+#else
+        verifyex(getConfigurationDirectory(nullptr, "temp", nullptr, "debug", debugDir));
+#endif
+        recursiveCreateDirectory(debugDir);
+    }
+public:
+    virtual void handle(ExceptionInterceptClass eClass, IException *e) const override
+    {
+        unsigned __int64 key = (byte)eClass << 24 | e->errorCode();
+        auto it = interceptHandlers.find(key);
+        if (it != interceptHandlers.end())
+            it->second(e);
+    }
+    virtual void setIntercept(ExceptionInterceptClass eClass, int code, InterceptHandler handler) override
+    {
+        unsigned __int64 key = (byte)eClass << 24 | code;
```

Comment:
Should this be testing the class, or should it only use the error code?  Is the issue that error code for os exceptions is the errno?

Reply:
does this relate to the mask suggestion?
I'm now filtering handling based on a mask (see new commit)